quinn-udp/benches/throughput.rs

```diff
@@ -27,8 +27,13 @@ pub fn criterion_benchmark(c: &mut Criterion) {
     // Reverse non-blocking flag set by `UdpSocketState` to make the test non-racy
     recv.set_nonblocking(false).unwrap();
 
-    let mut receive_buffer = vec![0; MAX_BUFFER_SIZE];
-    let mut meta = RecvMeta::default();
+    let gro_segments = UdpSocketState::new((&send).into()).unwrap().gro_segments();
+    let mut receive_buffers = vec![[0; SEGMENT_SIZE]; gro_segments];
+    let mut receive_slices = receive_buffers
+        .iter_mut()
+        .map(|buf| IoSliceMut::new(buf))
+        .collect::<Vec<_>>();
+    let mut meta = vec![RecvMeta::default(); gro_segments];
```

Comment:
This change seems to be confusing GRO with multi-message receive. GRO batches are passed to userspace as a single packet; that's why we used a single large buffer. By switching to many small buffers, you've actually disabled GRO.

Reply:
Apple doesn't **do** GSO/GRO, at least not via the socket interface. The `_x` calls are the semantic equivalent of the Linux `mmsg` calls. I'm borrowing the GSO/GRO helpers set up multi-message TX/RX in the bench.

Reply:
Sure, but this isn't an Apple-specific test; it runs on all platforms quinn-udp supports, some of which do have real GRO/GSO. This change will undermine performance testing on those platforms.

Reply:
This changed to
```rust
    let mut meta = vec![RecvMeta::default(); BATCH_SIZE];
```
in the meantime; is this still a concern?

(Because it seems that for `recvmmsg` platforms, you already needed that change?)

Reply:
> This changed to
> 
> ```rust
>     let mut meta = vec![RecvMeta::default(); BATCH_SIZE];
> ```

:+1: Thanks.

> in the meantime; is this still a concern?
> 
> (Because it seems that for `recvmmsg` platforms, you already needed that change?)

I initially wrote this benchmark (https://github.com/quinn-rs/quinn/pull/1915) to test GSO/GRO only. I am in favor of your addition, namely to leverage `recvmmsg` as well, i.e. pass more than one `iovec` to the OS to read into. Thanks @larseggert. I assume that on MacOS this benchmark now returns the same result for both `gro_result` `true` and `false`, correct?

To avoid any confusion, other than MacOS and iOS, I don't think there are any supported platforms that do `recvmmsg` but not `GRO`.

Linux / Android does [recvmmsg](https://man7.org/linux/man-pages/man2/recvmmsg.2.html) with GRO,  and Windows does [WSARecvMsg](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/ms741687(v=vs.85)) (equivalent to `recvmmsg`) with GRO. The BSDs do neither `recvmmsg` nor `GRO`, at least as implemented in `quinn-udp`.

The only exception now, with this pull request, is MacOS and iOS with `recvmsg_x`, acting like `recvmmsg`, but not supporting GRO.